### ISSUE TITLE
ClassCastException in Eclipse

### ISSUE THREAD

**Issue by Bert-R** [springfox/springfox#1054]
With Swagger2 enabled, I get the below class cast exception when running the application in Eclipse.

Strange enough, this exception does not occur when running the application from Gradle.

I'm running SpringFox 2.2.2 and Jackson 2.6.3.
See [``build.gradle``](https://github.com/yonadev/yona-server/blob/master/appservice/build.gradle)
Your help is appreciated!

```
java.lang.ClassCastException: com.fasterxml.classmate.members.ResolvedConstructor cannot be cast to com.fasterxml.classmate.members.ResolvedParameterizedMember
	at springfox.documentation.schema.property.FactoryMethodProvider$1.apply(FactoryMethodProvider.java:56)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:652)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators.tryFind(Iterators.java:752)
	at com.google.common.collect.Iterables.tryFind(Iterables.java:675)
	at com.google.common.collect.FluentIterable.firstMatch(FluentIterable.java:236)
	at springfox.documentation.schema.property.FactoryMethodProvider.in(FactoryMethodProvider.java:52)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.fromFactoryMethod(OptimizedModelPropertiesProvider.java:302)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.candidateProperties(OptimizedModelPropertiesProvider.java:191)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.propertiesFor(OptimizedModelPropertiesProvider.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
	at springfox.documentation.spring.web.caching.CachingAspect.cachedValue(CachingAspect.java:86)
	at springfox.documentation.spring.web.caching.CachingAspect.operationsAndProperties(CachingAspect.java:69)
	at sun.reflect.GeneratedMethodAccessor37.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:68)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy77.propertiesFor(Unknown Source)
	at springfox.documentation.schema.DefaultModelProvider.properties(DefaultModelProvider.java:151)
	at springfox.documentation.schema.DefaultModelProvider.modelFor(DefaultModelProvider.java:85)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:190)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:157)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:85)
	at springfox.documentation.spring.web.caching.CachingAspect.cachedValue(CachingAspect.java:86)
	at springfox.documentation.spring.web.caching.CachingAspect.modelsAndDependencies(CachingAspect.java:80)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:621)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:610)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:68)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:168)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:92)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
	at com.sun.proxy.$Proxy78.modelFor(Unknown Source)
	at springfox.documentation.spring.web.scanners.ApiModelReader.read(ApiModelReader.java:66)
	at springfox.documentation.spring.web.scanners.ApiListingScanner.scan(ApiListingScanner.java:91)
	at springfox.documentation.spring.web.scanners.ApiDocumentationScanner.scan(ApiDocumentationScanner.java:62)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.scanDocumentation(DocumentationPluginsBootstrapper.java:101)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:87)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:50)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:151)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:128)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:331)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:775)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:139)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:483)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:117)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:689)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:969)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:958)
	at nu.yona.server.AppServiceApplication.main(AppServiceApplication.java:36)
```

**Comment by dilipkrish** [comment]
Would you mind giving 2.2.3-SNAPSHOT a whirl and see if the problem is fixed?

**Comment by Bert-R** [comment]
With 2.2.3-SNAPSHOT, the stacktrace is different but the issue still exists:
```
com.google.common.util.concurrent.UncheckedExecutionException: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: com.fasterxml.classmate.members.ResolvedConstructor cannot be cast to com.fasterxml.classmate.members.ResolvedParameterizedMember
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4830)
	at springfox.documentation.schema.CachingModelProvider.modelFor(CachingModelProvider.java:56)
	at springfox.documentation.spring.web.scanners.ApiModelReader.read(ApiModelReader.java:67)
	at springfox.documentation.spring.web.scanners.ApiListingScanner.scan(ApiListingScanner.java:88)
	at springfox.documentation.spring.web.scanners.ApiDocumentationScanner.scan(ApiDocumentationScanner.java:69)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.scanDocumentation(DocumentationPluginsBootstrapper.java:101)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:87)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:50)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:151)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:128)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:331)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:775)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:139)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:483)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:117)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:689)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:969)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:958)
	at nu.yona.server.AppServiceApplication.main(AppServiceApplication.java:36)
Caused by: com.google.common.util.concurrent.UncheckedExecutionException: java.lang.ClassCastException: com.fasterxml.classmate.members.ResolvedConstructor cannot be cast to com.fasterxml.classmate.members.ResolvedParameterizedMember
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2203)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3937)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3941)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4824)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4830)
	at springfox.documentation.schema.property.CachingModelPropertiesProvider.propertiesFor(CachingModelPropertiesProvider.java:59)
	at springfox.documentation.schema.DefaultModelProvider.properties(DefaultModelProvider.java:151)
	at springfox.documentation.schema.DefaultModelProvider.modelFor(DefaultModelProvider.java:84)
	at springfox.documentation.schema.CachingModelProvider$1.load(CachingModelProvider.java:49)
	at springfox.documentation.schema.CachingModelProvider$1.load(CachingModelProvider.java:47)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
	... 23 common frames omitted
Caused by: java.lang.ClassCastException: com.fasterxml.classmate.members.ResolvedConstructor cannot be cast to com.fasterxml.classmate.members.ResolvedParameterizedMember
	at springfox.documentation.schema.property.FactoryMethodProvider$1.apply(FactoryMethodProvider.java:56)
	at com.google.common.collect.Iterators$7.computeNext(Iterators.java:652)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at com.google.common.collect.Iterators.tryFind(Iterators.java:752)
	at com.google.common.collect.Iterables.tryFind(Iterables.java:675)
	at com.google.common.collect.FluentIterable.firstMatch(FluentIterable.java:236)
	at springfox.documentation.schema.property.FactoryMethodProvider.in(FactoryMethodProvider.java:52)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.fromFactoryMethod(OptimizedModelPropertiesProvider.java:300)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.candidateProperties(OptimizedModelPropertiesProvider.java:189)
	at springfox.documentation.schema.property.OptimizedModelPropertiesProvider.propertiesFor(OptimizedModelPropertiesProvider.java:123)
	at springfox.documentation.schema.property.CachingModelPropertiesProvider$1.load(CachingModelPropertiesProvider.java:52)
	at springfox.documentation.schema.property.CachingModelPropertiesProvider$1.load(CachingModelPropertiesProvider.java:50)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3527)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2319)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2197)
	... 36 common frames omitted
```

**Comment by dilipkrish** [comment]
Do you have an example "DTO" that is causing this bug. Will help narrow down the issue.

Also another thing to check is that you have the correct version of classmate loaded in the class path.

**Comment by Bert-R** [comment]
What version of classmate should I have?

Groetjes, Bert

**Comment by dilipkrish** [comment]
[1.2.0](https://github.com/springfox/springfox/blob/master/gradle/dependencies.gradle#L11)

**Comment by Bert-R** [comment]
Thanks. I will check it tomorrow.

Groetjes, Bert

**Comment by Bert-R** [comment]
I have checked it and we are running 1.2.0. You asked what DTO causes this bug. I don't think it's DTO-specific. It occurs for all of our services. The simplest one is the adminservice. To reproduce it, you could checkout [the project](https://github.com/yonadev/yona-server), run ``gradlew eclipse``, open the project ``yona-server`` with its subprojects and and try to run ``nu.yona.server.AdminServiceApplication`` from the project ``adminservice``. That fails. The other two services fail too, all in the same way.

**Comment by dilipkrish** [comment]
super useful, let me dig into your project

**Comment by dilipkrish** [comment]
This has been fixed via 6e841d2551b4594f9e7faa6c4f3c20c7554a0e01

**Comment by Bert-R** [comment]
This indeed doesn't occur with the 2.2.3 snapshot anymore. Thanks!
